src/Refactoring-Core/RBInlineTemporaryRefactoring.class.st

```diff
@@ -38,55 +38,47 @@ RBInlineTemporaryRefactoring class >> model: aRBSmalltalk inline: anInterval fro
 ]
 
 { #category : 'preconditions' }
-RBInlineTemporaryRefactoring >> applicabilityPreconditions [
-
+RBInlineTemporaryRefactoring >> applicabilityPreconditions [ 
+	
 	^ {
-		  (RBCondition definesSelector: selector in: class).
-		  (RBCondition withBlock: [
-			   self checkSelectedInterval.
-			   true ]) }
-]
-
-{ #category : 'preconditions' }
-RBInlineTemporaryRefactoring >> checkSelectedInterval [
-
-	sourceTree := class parseTreeForSelector: selector.
-	sourceTree ifNil: [ self refactoringError: 'Could not parse source' ].
-	assignmentNode := sourceTree whichNodeIsContainedBy: sourceInterval.
-	assignmentNode isAssignment
-		ifFalse: [ self refactoringError: 'The selected node is not an assignment statement' ].
-	definingNode := assignmentNode whoDefines:
-		                assignmentNode variable name.
-	self hasOnlyOneAssignment
-		ifFalse: [ self refactoringError: 'There are multiple assignments to the variable' ].
-	( RBReadBeforeWrittenTester
-		isVariable: assignmentNode variable name
-		writtenBeforeReadIn: definingNode )
-		ifFalse: [ self refactoringError: 'The variable is possible read before it is assigned' ]
+		(ReDefinesSelectorsCondition new definesSelectors: (Array with: selector) in: class).
+		(ReIsValidInstanceVariableName new name: assignmentNode variable name; check).
+		(ReBlockCondition new 
+			block: [
+				(ReSingleAssignmentCondition new 
+					initialize: assignmentNode name parseTree: self parseTreeSearcher; check).
+				(ReVariablesNotReadBeforeWrittenCondition new 
+					variables: {assignmentNode variable name}; subtree: sourceTree; check).
+				true]).
+	}
```

Comment:
Now that we have all preconditions encapsulated we don't have the need for `ReBlockCondition` here anymore. So you can just remove it and add these two conditions to the list (instead of `ReBlockCondition`) with the rest of preconditions.